src/monocle_apptrace/wrap_common.py

```diff
@@ -172,17 +172,20 @@ def update_llm_endpoint(curr_span: Span, instance):
         if 'temperature' in instance.__dict__:
             temp_val = instance.__dict__.get("temperature")
             curr_span.set_attribute("temperature", temp_val)
-            # handling for model name
-        model_name = resolve_from_alias(instance.__dict__ , ["model","model_name"])
-        curr_span.set_attribute("model_name", model_name)
+        # handling for model name
+        model_name = resolve_from_alias(instance.__dict__, ["model", "model_name"])
+        if model_name:
+            curr_span.set_attribute("model_name", model_name)
         set_provider_name(curr_span, instance)
         # handling AzureOpenAI deployment
-        deployment_name = resolve_from_alias(instance.__dict__ , [ "engine", "azure_deployment",
-                                                                   "deployment_name", "deployment_id", "deployment"])
-        curr_span.set_attribute("az_openai_deployment", deployment_name)
+        deployment_name = resolve_from_alias(instance.__dict__, ["engine", "azure_deployment",
+                                                                 "deployment_name", "deployment_id", "deployment"])
+        if deployment_name:
```

Comment:
instead of `if` check, you can create a lambda like below

```
__set_span_attribute_if_not_none = lambda span, **kwargs: span.set_attribute(**{k: v for k, v in kwargs.items() if v is not None})

kwargs = {'arg1': some_value1, 'arg2': None, 'arg3': some_value3}
__set_span_attribute_if_not_none(curr_span, **kwargs)

```

Reply:
done @oi-raanne